Lib/email/headerregistry.py

```diff
@@ -369,8 +369,8 @@ def groups(self):
     @property
     def addresses(self):
         if self._addresses is None:
-            self._addresses = tuple(address for group in self._groups
-                                            for address in group.addresses)
+            self._addresses = tuple([address for group in self._groups
```

Comment:
The list comprehension is not necessary and doesn't improve the code.  This should stay as-is.

Reply:
The generator expression is slower. If you are fine with tiny performance regression introduced by the previous change I'll keep it.